api/src/test/java/edu/wpi/first/shuffleboard/api/util/UtilityClassTest.java

```diff
@@ -0,0 +1,35 @@
+package edu.wpi.first.shuffleboard.api.util;
+
+import org.junit.jupiter.api.Test;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@SuppressWarnings("PMD.AbstractClassWithoutAbstractMethod")
+public abstract class UtilityClassTest<T> {
+
+  private final Class<T> clazz;
+
+  public UtilityClassTest(Class<T> clazz) {
+    this.clazz = clazz;
+  }
```

Comment:
You could simplify this constructor by removing the constructor argument and doing this instead:
```java
public UtilityClassTest() {
    this.clazz = (new TypeToken<T>(getClass()) {}).getRawType();
}
```